tidepods/cli.py

```diff
@@ -1,50 +1,97 @@
 import click
 from datetime import datetime as dt
 
+
 def click_callback(f):
     return lambda _, __, x: f(x)
-    
+
+
 def check_date(date):
     try:
-        return dt.strptime(date , "%Y%m%d")
+        return dt.strptime(date, "%Y%m%d")
     except ValueError:
         msg = "Not a valid date: '{0}'.".format(date)
         raise Exception(msg)
-        
+
+
+def check_time(time):
+    try:
+        return dt.strptime(time, "%H:%M")
+    except ValueError:
+        msg = "Not a valid timestamp: '{0}'.".format(time)
+        raise Exception(msg)
+
+
 @click.group()
 def cli():
     """Tidal Surface Processing Tasks"""
     pass
 
+
 @cli.command()
-@click.option('-s','--shp', type=click.Path(dir_okay=False), required=True)
+@click.option('-i', '--infile', type=click.Path(dir_okay=False), required=True)
 @click.option('-t', '--template', type=click.Path(dir_okay=False), required=True)
-@click.option('-d', '--date', type=click.STRING, required=True, callback=click_callback(check_date))
-@click.argument('pfs', type=click.Path(dir_okay=False))
-
+@click.option('-d', '--date', type=click.STRING, required=True,
```

Comment:
It would be cleaner to implement `date` as a custom click type instead of this callback magic:

```python
class DateType(click.ParamType):
    name = 'date'

    def convert(self, value, _, __):
        return dt.strptime(date, "%Y%m%d")
```

You can then pass this as the `type` argument, and delete that `click_callback` thingy.

Reply:
Done